feat: support Apple Watch without Touch ID

krishukr opened the pull request:
Fix #4 

Reference:
https://developer.apple.com/documentation/localauthentication/lapolicy/deviceownerauthentication

Credits to:
https://github.com/jorgelbg/pinentry-touchid/issues/4#issuecomment-1030904964
https://github.com/lox/go-touchid

Requesting a new release.

krishukr committed:
use LAPolicyDeviceOwnerAuthentication

krishukr committed:
trigger build

krishukr committed:
update go releaser

krishukr committed:
update go releaser file

krishukr committed:
pin go releaser version

krishukr committed:
revert args of go releaser

GitSquared commented:
would love to have this, it's also really useful when docking the macbook and touchid is not available.

@jorgelbg doesn't look active on this repo anymore though. Maybe we should publish a fork...

krishukr commented:
Hi all, I noticed that there are people still looking for a solution to #4, and a while ago I wrote some arbitrary code to make it work. At that time, I downloaded the artifact from GitHub Actions CI to get a pre-built binary, but now it's expired. 

Considering the license and the lack of response from the author, I've created a Release [here](https://github.com/krishukr/pinentry-touchid/releases/tag/v0.0.4), which includes pre-built binaries and does not provide any warranty. 

You can follow the [manual installation](https://github.com/jorgelbg/pinentry-touchid#manual-installation) section from the readme file as instructions.

GitSquared commented:
@krishukr awesome, thanks!

<details>
<summary>Here's how I set it up to replace the homebrew-installed version:</summary>

- Download [this edited formula file](https://github.com/GitSquared/dotfiles/blob/neuromancer/.homebrew-formulas/pinentry-touchid.rb). Save it somewhere e.g `~/.homebrew-formulas/pinentry-touchid.rb`
- Untap the mainline version: `brew untap jorgelbg/tap`
- Install the custom formula: `brew install --formula ~/.homebrew-formulas/pinentry-touchid.rb`
- done!

</details>

krishukr committed:
merge 'MaxNeedsSnacks' into 'krishukr'

Squashed commit of the following:

commit 55a8866f809a88317bc850f9b9d8a7da9c53e546
Author: max. <max@someone.ky>
Date:   Fri May 23 18:04:26 2025 +0200

    add curses fallback support

commit bf0d7fc892cc75f45900d11bc3d0791733d9e414
Author: Ethan P <eth-p+git@hidden.email>
Date:   Mon Apr 24 14:41:08 2023 -0700

    Use `pinentry-mac` directly if TouchID unavailable

    By using execve(2), we can replace the running `pinentry-touchid`
    process with `pinentry-mac`. This allows it to directly communicate with
    the GPG agent, receiving all the pinentry settings and messages.

commit 54d4d0e1d91a9986eefc42192208a50957b9f701
Author: max. <max@someone.ky>
Date:   Fri May 23 17:26:26 2025 +0200

    use local fork of go-touchid

commit c36dbd41277d27568a60e712233ff39a1668fbdc
Author: Weiyi Chen <wychen325@gmail.com>
Date:   Sat Feb 10 07:32:20 2024 +0800

    feat: update README

commit 67118d7743eec27faac147ec14250d5b6181d4ff
Author: Weiyi Chen <wychen325@gmail.com>
Date:   Sat Feb 10 06:49:06 2024 +0800

    fix: support password authentication

krishukr committed:
fix(ci): upgrade actions/upload-artifact v3 -> v4

CasualDeveloper commented:
> Hi all, I noticed that there are people still looking for a solution to #4, and a while ago I wrote some arbitrary code to make it work. At that time, I downloaded the artifact from GitHub Actions CI to get a pre-built binary, but now it's expired.
> 
> Considering the license and the lack of response from the author, I've created a Release [here](https://github.com/krishukr/pinentry-touchid/releases/tag/v0.0.4), which includes pre-built binaries and does not provide any warranty.
> 
> You can follow the [manual installation](https://github.com/jorgelbg/pinentry-touchid?rgh-link-date=2025-04-04T07%3A50%3A23Z#manual-installation) section from the readme file as instructions.

I think this is worth turning into a proper fork lol. maybe rename the repo `pinentry-watchid` ? :)
